src/server/lock_manager/config.rs

```diff
@@ -3,26 +3,47 @@
 use super::deadlock::Scheduler as DeadlockScheduler;
 use super::waiter_manager::Scheduler as WaiterMgrScheduler;
 use configuration::{rollback_or, ConfigChange, ConfigManager, Configuration, RollbackCollector};
+use serde::de::{Deserialize, Deserializer};
 
 use std::error::Error;
+use tikv_util::config::{rduration_deser_from_str, ReadableDuration};
 
 #[derive(Clone, Serialize, Deserialize, PartialEq, Debug, Configuration)]
 #[serde(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct Config {
     #[config(skip)]
     pub enabled: bool,
-    pub wait_for_lock_timeout: u64,
-    pub wake_up_delay_duration: u64,
+    #[serde(deserialize_with = "compat_desr")]
+    pub wait_for_lock_timeout: ReadableDuration,
+    #[serde(deserialize_with = "compat_desr")]
+    pub wake_up_delay_duration: ReadableDuration,
     pub pipelined: bool,
 }
 
+#[derive(Deserialize)]
+#[serde(untagged)]
+enum StrOrNum<'a> {
+    Str(&'a str),
+    Num(u64),
+}
+
+fn compat_desr<'de, D>(deserializer: D) -> Result<ReadableDuration, D::Error>
```

Comment:
How about it?
```rust
// u64 is for backward compatibility since v3.x uses it.
fn readable_duration_or_u64<'de, D>(deserializer: D) -> Result<ReadableDuration, D::Error>
where
    D: Deserializer<'de>,
    D::Error: serde::de::Error,
{
    use serde_json::Value;

    let v = Value::deserialize(deserializer)?;
    match v {
        Value::String(s) => ReadableDuration::deserialize(s.into_deserializer()),
        Value::Number(n) => n
            .as_u64()
            .map(|n| ReadableDuration::millis(n))
            .ok_or_else(|| D::Error::custom(format!("expect unsigned integer: {}", n))),
        other => Err(D::Error::custom(format!(
            "expect ReadableDuration or unsigned integer: {}",
            other
        ))),
    }
}
```